### ISSUE TITLE
faster travis builds using containers

### ISSUE THREAD

**Issue by piyush-kurur** [diagrams/monoid-extras#25]
Travis build are faster if one uses the container infrastructure. The
main limitation is that one is not allowed to use sudo. However, hvr's
ppa is available for the container infrastructure See:
- https://docs.travis-ci.com/user/workers/container-based-infrastructure/
- https://github.com/hvr/multi-ghc-travis#travisyml-for-container-based-infrastructure

It might be a good idea to move to this style.

**Comment by bergey** [comment]
Yes, we should move `monoid-extras` and all the Diagrams packages to containers.  If you have time to make a PR, that would be great.

My last attempt is at https://github.com/diagrams/diagrams-core/compare/travis.  I ran out of time to figure out why the build was failing.  `monoid-extras` should be easier because it has no dependencies that need to track git HEAD.
